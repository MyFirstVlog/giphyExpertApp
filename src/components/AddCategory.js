import React, {useState} from 'react'
import PropTypes from 'prop-types'

export const AddCategory = ({setCategorias}) => {

    const [inputValue, setInputValue] = useState('')

    const handleChange = (e)=>{
        setInputValue(e.target.value)
    }

    const handleSubmit = (e) => {
        e.preventDefault()

        if(inputValue.trim().length >= 2){
            setCategorias((actuales) => [inputValue,...actuales] )
            setInputValue('')
        }
    }

    return (
        <form onSubmit= {handleSubmit}>
            <input
                placeholder = 'Que desea buscar?'
                type="text"
                value={inputValue}
                onChange={handleChange}
            >
            </input>
        </form>
    )
}

AddCategory.propTypes = {
    setCategorias : PropTypes.func.isRequired
}