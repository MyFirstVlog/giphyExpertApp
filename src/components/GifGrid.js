import React from 'react'
import PropTypes from 'prop-types'
import { getGifs } from '../helpers/GetGifs'
import { useFecthGifs } from '../hooks/useFechGifs'
import { GifGridItem } from './GifGridItem'

export const GifGrid = ({ category }) => {

    const {data:images,loading} = useFecthGifs(category)

    /**
     * recordar que gifs => setimages(gifs) -> setimages
     * 
     * <GifGridItem key={img.id} {...img} --> manda la info de img como props independientes
     * 
     * const {data:images,loading} -> para renombrar variables
     */



    return (
        <>
            <h3>{category}</h3>

            {loading && <p className='animate__animated animate__jackInTheBox'>Loading...</p>}

            <div className='card-grid'>
                 {images.map(img => {
                     return <GifGridItem key={img.id} {...img} />
                })}
            </div>
        </>
    )
}

GifGrid.propTypes = {
    category : PropTypes.string.isRequired
} 