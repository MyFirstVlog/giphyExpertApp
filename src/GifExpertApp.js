import React, {useState} from 'react';
import { AddCategory } from './components/AddCategory';
import { GifGrid } from './components/GifGrid';

const categoriasRandom = ['Parasite', 'One Punch', 'Tokyo Ghoul']
let init = ['Naruto', 'Attack On Titans', 'Kimetzu no Yaiba']

const GifExpertApp = ({defaultCategories = []}) => {


    const [categorias, setCategorias] = useState(defaultCategories)


    // const handleAdd = () => {
    //     const categoriasRandomNro = Math.floor(Math.random()*3)
    //     const valorAPoner = categoriasRandom[categoriasRandomNro]
    //     setCategorias([valorAPoner,...categorias])
    // }

    return(
        <>
            <h2>GifExpertApp</h2>

            <AddCategory setCategorias={setCategorias}/>
            <hr/>

            <ol>
                {
                    categorias.map((each) =>
                    <GifGrid 
                        key={each}
                        category={each}
                    />)
                }
            </ol>
        </>
    )
} 

export default GifExpertApp