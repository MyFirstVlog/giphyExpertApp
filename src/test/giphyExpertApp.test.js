import React from 'react';
import { shallow } from 'enzyme';
import GifExpertApp from '../GifExpertApp'
import { GifGridItem } from '../components/GifGridItem';

describe('Pruebas a la aplicacion del giphy', () => {

    const url = 'https://localhost.com'
    const title = 'La vie de Rick et Morty'
    
    let  wrapper = shallow(<GifGridItem url= {url} title={title} />)
    
    test('debe mostrar el componente gifGridItem correctamente', () => {

        expect(wrapper).toMatchSnapshot()
    })

    test('debe de tener un parrafo con el title', () => {
        
        const porcionDePhtml = wrapper.find('p').text()

        expect(porcionDePhtml.trim()).toBe(title)

    })

    test('debe tener la img en las props url y el alt de la imagen', () => {

        const propsImg =wrapper.find('img')
        expect(propsImg.prop('src')).toBe(url)
        expect(propsImg.prop('alt')).toBe(title)
        
    })

    test('debe de tener el fade bounce', () => {
        
        const propsDiv =wrapper.find('div')

        expect(propsDiv.prop('className').includes('animate__bounce')).toBe(true)

    })
    
    
    
    
}) 
