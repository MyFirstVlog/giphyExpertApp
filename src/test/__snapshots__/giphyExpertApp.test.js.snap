// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas a la aplicacion del giphy debe mostrar el componente gifGridItem correctamente 1`] = `
<div
  className="card animate__animated animate__bounce"
>
  <img
    alt="La vie de Rick et Morty"
    src="https://localhost.com"
  />
  <p>
    La vie de Rick et Morty
  </p>
</div>
`;
